src/src/Form/CustomerSyncFormDataConfiguration.php

```diff
@@ -0,0 +1,137 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * Smaily for PrestaShop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Smaily for PrestaShop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Smaily for PrestaShop. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Smaily <info@smaily.com>
+ * @copyright 2024 Smaily
+ * @license   GPL3
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\Module\SmailyForPrestaShop\Form;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+final class CustomerSyncFormDataConfiguration implements DataConfigurationInterface
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    public function __construct(ConfigurationInterface $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function getConfiguration(): array
+    {
+        $return = [];
+
+        $return['enabled'] = $this->configuration->getBoolean('SMAILY_ENABLE_CUSTOMER_SYNC');
+        $return['sync_additional'] = json_decode($this->configuration->get('SMAILY_SYNCRONIZE_ADDITIONAL'), true);
+        $return['cron_token'] = $this->configuration->get('SMAILY_CUSTOMER_CRON_TOKEN');
+        $return['cron_url'] = $this->buildCronURL($return['cron_token']);
+        $return['optin_enabled'] = $this->configuration->getBoolean('SMAILY_OPTIN_ENABLED');
+        $return['autoresponder'] = $this->configuration->get('SMAILY_OPTIN_AUTORESPONDER');
+
+        return $return;
+    }
+
+    public function updateConfiguration(array $formData): array
+    {
+        $errors = [];
+
+        $syncEnabled = $formData['enabled'];
+        $additionalFields = $formData['sync_additional'];
+        $optInEnabled = $formData['optin_enabled'];
+        $cronToken = $formData['cron_token'];
+        $autoresponder = $formData['autoresponder'];
+
+        if (empty($cronToken)) {
+            $errors[] = [
+                'key' => 'Please provide a cron token for customer synchronization.',
+                'domain' => 'Modules.Smailyforprestashop.Admin',
+                'parameters' => [],
+            ];
+        }
+
+        if ($optInEnabled && empty($autoresponder)) {
+            $errors[] = [
+                'key' => 'Please select an automation workflow for customer Opt-In trigger.',
+                'domain' => 'Modules.Smailyforprestashop.Admin',
+                'parameters' => [],
+            ];
+        }
```

Comment:
Again, not sure of the scope of this release, but modules for other platforms allow saving empty autoresponder to allow triggering `opt-in` automation workflows.

Reply:
Hmm.. if I understand correctly then this is for cases when users don't want to send out opt-in letters or have other logic behind the opt-in trigger. They just want to add users to the subscribers list.

Reply:
Modules for other platforms are using the logic I described - when no automation workflow is selected, then it is treated as an opt-in trigger. I think we should stay in line with other modules for consistency.

If the user has different logic behind opt-in trigger, then they would need to use form submitted trigger. And if they would just want to add users to subscribers list, then they can use customer synchronization. Capturing newsletter signups is intended for opting in subscriber in Smaily and triggering welcome automation workflow.

Reply:
Refactored a bit and made choosing the trigger optional. https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/bbde0cd85aedb234f7e75c7c388e0be3df7509ec